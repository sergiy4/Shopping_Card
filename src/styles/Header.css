.header_stick{
    
    font-family: 'Raleway', sans-serif;

    background-color: rgb(37, 255, 164);
    color:rgb(11, 173, 87);
    height: 15%;
    display: flex;
    align-items: center;
    justify-content: space-around;
    transform-style: preserve-3d;
    
}
.name_shop{
    cursor: pointer;
    
}
.list{
    list-style: none;
    display: flex;
    gap:40px;
       
}