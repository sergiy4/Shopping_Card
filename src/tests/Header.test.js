import {render, screen,cleanup} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Header from '../components/HeaderComponent'
import "@testing-library/jest-dom"

afterEach(()=>{
    cleanup()
})

describe('testing Header component', ()=>{
    test('correct render',()=>{
        const {asFragment} = render(<Header/>)

        expect(asFragment()).toMatchSnapshot()

    })

    test('check correctly render h1',()=>{
        render(<Header/>)

        const h1 = screen.getByText('FakePlants')

        expect(h1).toBeInTheDocument()
    })

    test('check correctly render list',()=>{
        render(<Header/>)

        const list = screen.getByRole('list')
        const listItems = screen.getAllByRole('listitem')

        expect(list).toBeInTheDocument()
        expect(listItems.length).toEqual(3)
    })
    

})