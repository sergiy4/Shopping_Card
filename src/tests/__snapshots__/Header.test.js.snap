// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header component correct render 1`] = `
<DocumentFragment>
  <header>
    <h1>
      FakePlants
    </h1>
    <ul>
      <li>
        Shop
      </li>
      <li>
        Contact
      </li>
      <li>
        Shopping cart
      </li>
    </ul>
  </header>
</DocumentFragment>
`;
